Modules/Loadable/Markups/qSlicerMarkupsModuleWidget.cxx

```diff
@@ -2403,6 +2450,61 @@ void qSlicerMarkupsModuleWidget::onResetToDefaultDisplayPropertiesPushButtonClic
   this->markupsLogic()->SetDisplayNodeToDefaults(displayNode);
 }
 
+//-----------------------------------------------------------------------------
+void qSlicerMarkupsModuleWidget::onapplyCurveResamplingPushButtonClicked()
+{
+  Q_D(qSlicerMarkupsModuleWidget);
+
+  double resampleRate = d->SelectNumberOfSamplePointsSpinBox->value();
+
+  vtkMRMLMarkupsCurveNode *markupsCurveNode = vtkMRMLMarkupsCurveNode::SafeDownCast(d->MRMLNodeComboBox_inputNodeSelector->currentNode());
+
+  if (!markupsCurveNode)
+    {
+	vtkMRMLMarkupsClosedCurveNode *markupsCurveNode = vtkMRMLMarkupsClosedCurveNode::SafeDownCast(d->MRMLNodeComboBox_inputNodeSelector->currentNode());
```

Comment:
use the markups curve node member variable instead of getting it again from the GUI